### PR TITLE
document to_interface, from_interface, create interface

### PR DESCRIPTION
**Pull request by paulbrauner-da**

### CODE REVIEW

File: docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst [hunk 1]
```diff
@@ -1612,9 +1571,127 @@ is the one defined in package ``dep-1.0.0``. Contract ``5678`` is thus
...
+                i : Optional Int
+              where
+                signatory p
+
+                interface instance I for T where
+                  view = IView {}
+
+Finally, assume that the module defining the first version of ``T`` is imported
+as ``V1``, and the module defining the second version of ``T`` is imported as
+``V2``. Then the expression ``fromInterface @V2.T (toInterface @I (V1.T 'Alice'))``
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst:1617]:
```suggestion
``V2``. The expression ``fromInterface @V2.T (toInterface @I (V1.T 'Alice'))``
```

File: docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst [hunk 2]
```diff
@@ -1612,9 +1571,127 @@ is the one defined in package ``dep-1.0.0``. Contract ``5678`` is thus
...
+
+Dynamic Target Template Selection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In a top-level exercise triggered by a Ledger API command, or in an interface
+fetch or exercise triggered from the body of a choice, the target template is
+determined at runtime by the rules of package preference detailed in the
+:ref:`dynamic package
+resolution<dynamic-package-resolution-in-command-submission>` section of the
+smart contract upgrade documentation.
```
Comment (carrielaben-da) [inline docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst:1635]:
```suggestion
fetch or exercise triggered from the body of a choice, the rules of package preference detailed in
:ref:`dynamic package
resolution<dynamic-package-resolution-in-command-submission>`  determine the target template at runtime.
```

**Review by carrielaben-da — APPROVED** [review]
Add to 3.x docs if necessary. Thanks!

File: docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst [hunk 3]
```diff
@@ -1453,87 +1450,49 @@ Below, we detail the rules governing target template selection, then explain how
...
+
+Daml contracts are represented by one of two sorts of values at runtime:
+template values or interface values.
+
+* Template values are those whose concrete template type is statically
+  known. They are obtained by directly constructing a template record, or by a
+  call to ``fetch``. Their runtime representation is a simple record.
+* Interface values are those whose concrete template type is not fully 
+  statically known, aside from the fact that it implements a given interface.
+  They are obtained by applying ``toInterface`` to a template value.
```
Comment (remyhaemmerle-da) [inline docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst:1474]: should we not say that this pair is not accessible by user ? I mean it should be consider as a blob.

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks.

File: docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst (hunk 3 above)
Comment (paulbrauner-da) [reply docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst:1474]: Good point, added this sentence:

Note that the type of interface values is opaque: while it is useful to
conceptualize interface values as pairs for defining the runtime semantics of
the language, their actual implementation may vary and is not exposed to the
user.

**Review by dylant-da — APPROVED** [review]

**Pull request merged**